crypto/pkcs12/p12_crt.c

```diff
@@ -69,12 +70,18 @@ PKCS12 *PKCS12_create_ex(const char *pass, const char *name, EVP_PKEY *pkey,
             goto err;
         if (keyidlen && !PKCS12_add_localkeyid(bag, keyid, keyidlen))
             goto err;
+        if (cb)
+            if (cb(bag, cbarg) == -1)
+                ERR_raise(ERR_LIB_PKCS12, PKCS12_R_CALLBACK_FAILED);
```

Comment:
In this case I think there should be a `goto err` at this point. Failure here means abandon the whole construction.